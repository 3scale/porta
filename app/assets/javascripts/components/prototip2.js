

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
