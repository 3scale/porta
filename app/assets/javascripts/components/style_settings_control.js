// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
