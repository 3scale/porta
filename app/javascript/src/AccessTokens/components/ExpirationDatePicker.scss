@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
