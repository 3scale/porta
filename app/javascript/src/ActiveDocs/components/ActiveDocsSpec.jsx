// @flow

import React from 'react'

import { createReactWrapper } from 'utilities/createReactWrapper'
import SwaggerUI from 'swagger-ui-react'

import 'swagger-ui-react/swagger-ui.css'

type ActiveDocsSpecProps = {
  url: string
}

const ActiveDocsSpec = ({ url }: ActiveDocsSpecProps) => <SwaggerUI url={url} />

const ActiveDocsSpecWrapper = (props: ActiveDocsSpecProps, id: string) => createReactWrapper(<ActiveDocsSpec {...props} />, id)

export { ActiveDocsSpec, ActiveDocsSpecWrapper }
