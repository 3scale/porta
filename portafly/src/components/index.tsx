export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
