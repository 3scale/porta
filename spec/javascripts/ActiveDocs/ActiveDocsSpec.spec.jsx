// @flow

import React from 'react'
import { mount } from 'enzyme'
import SwaggerUI from 'swagger-ui-react'
import { ActiveDocsSpec } from 'ActiveDocs/components/ActiveDocsSpec'

const url = 'foo.example.com'
const accountDataUrl = 'foo.example.com/foo.json'
const autocompleteOAS3 = jest.fn(() => Promise.resolve({}))

it('should render itself', () => {
  const wrapper = mount(<ActiveDocsSpec url={url} accountDataUrl={accountDataUrl} autocompleteOAS3={autocompleteOAS3} />)
  expect(wrapper.find(ActiveDocsSpec).exists()).toBe(true)
  expect(wrapper).toMatchSnapshot()
})

it('should render swagger-ui', () => {
  const wrapper = mount(<ActiveDocsSpec url={url} accountDataUrl={accountDataUrl} autocompleteOAS3={autocompleteOAS3}/>)
  expect(wrapper.find('.swagger-ui').exists()).toBe(true)
  expect(wrapper.find(SwaggerUI).prop('url')).toBe('foo.example.com')
  expect(typeof wrapper.find(SwaggerUI).prop('responseInterceptor')).toBe('function')
})
