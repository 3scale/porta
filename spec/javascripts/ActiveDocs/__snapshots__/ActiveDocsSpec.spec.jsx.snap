// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<ActiveDocsSpec
  responseInterceptor={[MockFunction]}
  url="foo.example.com"
>
  <SwaggerUI
    responseInterceptor={[MockFunction]}
    url="foo.example.com"
  >
    <r>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(r)>
          <r
            dispatch={[Function]}
            state={
              Immutable.Map {
                "layout": Immutable.Map {
                  "layout": "BaseLayout",
                  "filter": null,
                },
                "spec": Immutable.Map {
                  "spec": "",
                  "url": "foo.example.com",
                  "loadingStatus": "loading",
                },
                "configs": Immutable.Map {},
                "err": Immutable.Map {},
                "auth": Immutable.Map {},
                "oas3": Immutable.Map {},
              }
            }
          >
            <t
              Im={
                Object {
                  "Collection": [Function],
                  "Iterable": [Function],
                  "List": [Function],
                  "Map": [Function],
                  "OrderedMap": [Function],
                  "OrderedSet": [Function],
                  "Range": [Function],
                  "Record": [Function],
                  "Repeat": [Function],
                  "Seq": [Function],
                  "Set": [Function],
                  "Stack": [Function],
                  "fromJS": [Function],
                  "is": [Function],
                }
              }
              React={
                Object {
                  "Children": Object {
                    "count": [Function],
                    "forEach": [Function],
                    "map": [Function],
                    "only": [Function],
                    "toArray": [Function],
                  },
                  "Component": [Function],
                  "Fragment": Symbol(react.fragment),
                  "Profiler": Symbol(react.profiler),
                  "PureComponent": [Function],
                  "StrictMode": Symbol(react.strict_mode),
                  "Suspense": Symbol(react.suspense),
                  "__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED": Object {
                    "IsSomeRendererActing": Object {
                      "current": false,
                    },
                    "ReactComponentTreeHook": Object {},
                    "ReactCurrentBatchConfig": Object {
                      "suspense": null,
                    },
                    "ReactCurrentDispatcher": Object {
                      "current": Object {
                        "readContext": [Function],
                        "useCallback": [Function],
                        "useContext": [Function],
                        "useDebugValue": [Function],
                        "useDeferredValue": [Function],
                        "useEffect": [Function],
                        "useImperativeHandle": [Function],
                        "useLayoutEffect": [Function],
                        "useMemo": [Function],
                        "useReducer": [Function],
                        "useRef": [Function],
                        "useResponder": [Function],
                        "useState": [Function],
                        "useTransition": [Function],
                      },
                    },
                    "ReactCurrentOwner": Object {
                      "current": null,
                    },
                    "ReactDebugCurrentFrame": Object {
                      "getCurrentStack": null,
                      "getStackAddendum": [Function],
                    },
                    "assign": [Function],
                  },
                  "cloneElement": [Function],
                  "createContext": [Function],
                  "createElement": [Function],
                  "createFactory": [Function],
                  "createRef": [Function],
                  "forwardRef": [Function],
                  "isValidElement": [Function],
                  "lazy": [Function],
                  "memo": [Function],
                  "useCallback": [Function],
                  "useContext": [Function],
                  "useDebugValue": [Function],
                  "useEffect": [Function],
                  "useImperativeHandle": [Function],
                  "useLayoutEffect": [Function],
                  "useMemo": [Function],
                  "useReducer": [Function],
                  "useRef": [Function],
                  "useState": [Function],
                  "version": "16.12.0",
                }
              }
              auth={[Function]}
              authActions={
                Object {
                  "authorize": [Function],
                  "authorizeAccessCodeWithBasicAuthentication": [Function],
                  "authorizeAccessCodeWithFormParams": [Function],
                  "authorizeApplication": [Function],
                  "authorizeOauth2": [Function],
                  "authorizePassword": [Function],
                  "authorizeRequest": [Function],
                  "configureAuth": [Function],
                  "logout": [Function],
                  "preAuthorizeImplicit": [Function],
                  "showDefinitions": [Function],
                }
              }
              authSelectors={
                Object {
                  "authorized": [Function],
                  "definitionsForRequirements": [Function],
                  "definitionsToAuthorize": [Function],
                  "getConfigs": [Function],
                  "getDefinitionsByNames": [Function],
                  "isAuthorized": [Function],
                  "shownDefinitions": [Function],
                }
              }
              configs={Object {}}
              configsActions={
                Object {
                  "loaded": [Function],
                  "toggle": [Function],
                  "update": [Function],
                }
              }
              configsSelectors={
                Object {
                  "get": [Function],
                }
              }
              dispatch={[Function]}
              err={[Function]}
              errActions={
                Object {
                  "clear": [Function],
                  "clearBy": [Function],
                  "newAuthErr": [Function],
                  "newSpecErr": [Function],
                  "newSpecErrBatch": [Function],
                  "newThrownErr": [Function],
                  "newThrownErrBatch": [Function],
                }
              }
              errSelectors={
                Object {
                  "allErrors": [Function],
                  "lastError": [Function],
                }
              }
              fn={
                Object {
                  "buildRequest": [Function],
                  "createXMLExample": [Function],
                  "execute": [Function],
                  "fetch": [Function],
                  "getScrollParent": [Function],
                  "inferSchema": [Function],
                  "memoizedCreateXMLExample": [Function],
                  "memoizedSampleFromSchema": [Function],
                  "opId": [Function],
                  "opsFilter": [Function],
                  "resolve": [Function],
                  "resolveSubtree": [Function],
                  "sampleFromSchema": [Function],
                  "sampleXmlFromSchema": [Function],
                  "serializeRes": [Function],
                  "shallowEqualKeys": [Function],
                }
              }
              getComponent={[Function]}
              getComponents={[Function]}
              getConfigs={[Function]}
              getState={[Function]}
              getStore={[Function]}
              getSystem={[Function]}
              initOAuth={[Function]}
              layout={[Function]}
              layoutActions={
                Object {
                  "changeMode": [Function],
                  "clearScrollTo": [Function],
                  "parseDeepLinkHash": [Function],
                  "readyToScroll": [Function],
                  "scrollTo": [Function],
                  "scrollToElement": [Function],
                  "show": [Function],
                  "updateFilter": [Function],
                  "updateLayout": [Function],
                }
              }
              layoutSelectors={
                Object {
                  "current": [Function],
                  "currentFilter": [Function],
                  "getScrollToKey": [Function],
                  "isShown": [Function],
                  "isShownKeyFromUrlHashArray": [Function],
                  "showSummary": [Function],
                  "urlHashArrayFromIsShownKey": [Function],
                  "whatMode": [Function],
                }
              }
              log={[Function]}
              makeMappedContainer={[Function]}
              oas3={[Function]}
              oas3Actions={
                Object {
                  "setActiveExamplesMember": [Function],
                  "setRequestBodyValue": [Function],
                  "setRequestContentType": [Function],
                  "setResponseContentType": [Function],
                  "setSelectedServer": [Function],
                  "setServerVariableValue": [Function],
                }
              }
              oas3Selectors={
                Object {
                  "activeExamplesMember": [Function],
                  "requestBodyValue": [Function],
                  "requestContentType": [Function],
                  "responseContentType": [Function],
                  "selectedServer": [Function],
                  "serverEffectiveValue": [Function],
                  "serverVariableValue": [Function],
                  "serverVariables": [Function],
                }
              }
              preauthorizeApiKey={[Function]}
              preauthorizeBasic={[Function]}
              render={[Function]}
              spec={[Function]}
              specActions={
                Object {
                  "changeConsumesValue": [Function],
                  "changeParam": [Function],
                  "changeParamByIdentity": [Function],
                  "changeProducesValue": [Function],
                  "clearRequest": [Function],
                  "clearResponse": [Function],
                  "clearValidateParams": [Function],
                  "download": [Function],
                  "downloadConfig": [Function],
                  "execute": [Function],
                  "executeRequest": [Function],
                  "getConfigByUrl": [Function],
                  "invalidateResolvedSubtreeCache": [Function],
                  "logRequest": [Function],
                  "parseToJson": [Function],
                  "requestResolvedSubtree": [Function],
                  "resolveSpec": [Function],
                  "setMutatedRequest": [Function],
                  "setRequest": [Function],
                  "setResponse": [Function],
                  "setScheme": [Function],
                  "updateEmptyParamInclusion": [Function],
                  "updateJsonSpec": [Function],
                  "updateLoadingStatus": [Function],
                  "updateResolved": [Function],
                  "updateResolvedSubtree": [Function],
                  "updateSpec": [Function],
                  "updateUrl": [Function],
                  "validateParams": [Function],
                }
              }
              specSelectors={
                Object {
                  "allowTryItOutFor": [Function],
                  "basePath": [Function],
                  "canExecuteScheme": [Function],
                  "consumes": [Function],
                  "consumesOptionsFor": [Function],
                  "contentTypeValues": [Function],
                  "currentProducesFor": [Function],
                  "definitions": [Function],
                  "externalDocs": [Function],
                  "findDefinition": [Function],
                  "getLocalConfig": [Function],
                  "getParameter": [Function],
                  "hasHost": [Function],
                  "host": [Function],
                  "info": [Function],
                  "isOAS3": [Function],
                  "isSwagger2": [Function],
                  "lastError": [Function],
                  "loadingStatus": [Function],
                  "mutatedRequestFor": [Function],
                  "mutatedRequests": [Function],
                  "operationScheme": [Function],
                  "operationWithMeta": [Function],
                  "operations": [Function],
                  "operationsWithRootInherited": [Function],
                  "operationsWithTags": [Function],
                  "parameterInclusionSettingFor": [Function],
                  "parameterValues": [Function],
                  "parameterWithMeta": [Function],
                  "parameterWithMetaByIdentity": [Function],
                  "parametersIncludeIn": [Function],
                  "parametersIncludeType": [Function],
                  "paths": [Function],
                  "produces": [Function],
                  "producesOptionsFor": [Function],
                  "requestFor": [Function],
                  "requests": [Function],
                  "responseFor": [Function],
                  "responses": [Function],
                  "schemes": [Function],
                  "security": [Function],
                  "securityDefinitions": [Function],
                  "semver": [Function],
                  "servers": [Function],
                  "spec": [Function],
                  "specJson": [Function],
                  "specJsonWithResolvedSubtrees": [Function],
                  "specResolved": [Function],
                  "specResolvedSubtree": [Function],
                  "specSource": [Function],
                  "specStr": [Function],
                  "tagDetails": [Function],
                  "taggedOperations": [Function],
                  "tags": [Function],
                  "url": [Function],
                  "validateBeforeExecute": [Function],
                  "version": [Function],
                }
              }
              state={
                Immutable.Map {
                  "layout": Immutable.Map {
                    "layout": "BaseLayout",
                    "filter": null,
                  },
                  "spec": Immutable.Map {
                    "spec": "",
                    "url": "foo.example.com",
                    "loadingStatus": "loading",
                  },
                  "configs": Immutable.Map {},
                  "err": Immutable.Map {},
                  "auth": Immutable.Map {},
                  "oas3": Immutable.Map {},
                }
              }
            >
              <Connect(r)>
                <r
                  dispatch={[Function]}
                  state={
                    Immutable.Map {
                      "layout": Immutable.Map {
                        "layout": "BaseLayout",
                        "filter": null,
                      },
                      "spec": Immutable.Map {
                        "spec": "",
                        "url": "foo.example.com",
                        "loadingStatus": "loading",
                      },
                      "configs": Immutable.Map {},
                      "err": Immutable.Map {},
                      "auth": Immutable.Map {},
                      "oas3": Immutable.Map {},
                    }
                  }
                >
                  <t
                    Im={
                      Object {
                        "Collection": [Function],
                        "Iterable": [Function],
                        "List": [Function],
                        "Map": [Function],
                        "OrderedMap": [Function],
                        "OrderedSet": [Function],
                        "Range": [Function],
                        "Record": [Function],
                        "Repeat": [Function],
                        "Seq": [Function],
                        "Set": [Function],
                        "Stack": [Function],
                        "fromJS": [Function],
                        "is": [Function],
                      }
                    }
                    React={
                      Object {
                        "Children": Object {
                          "count": [Function],
                          "forEach": [Function],
                          "map": [Function],
                          "only": [Function],
                          "toArray": [Function],
                        },
                        "Component": [Function],
                        "Fragment": Symbol(react.fragment),
                        "Profiler": Symbol(react.profiler),
                        "PureComponent": [Function],
                        "StrictMode": Symbol(react.strict_mode),
                        "Suspense": Symbol(react.suspense),
                        "__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED": Object {
                          "IsSomeRendererActing": Object {
                            "current": false,
                          },
                          "ReactComponentTreeHook": Object {},
                          "ReactCurrentBatchConfig": Object {
                            "suspense": null,
                          },
                          "ReactCurrentDispatcher": Object {
                            "current": Object {
                              "readContext": [Function],
                              "useCallback": [Function],
                              "useContext": [Function],
                              "useDebugValue": [Function],
                              "useDeferredValue": [Function],
                              "useEffect": [Function],
                              "useImperativeHandle": [Function],
                              "useLayoutEffect": [Function],
                              "useMemo": [Function],
                              "useReducer": [Function],
                              "useRef": [Function],
                              "useResponder": [Function],
                              "useState": [Function],
                              "useTransition": [Function],
                            },
                          },
                          "ReactCurrentOwner": Object {
                            "current": null,
                          },
                          "ReactDebugCurrentFrame": Object {
                            "getCurrentStack": null,
                            "getStackAddendum": [Function],
                          },
                          "assign": [Function],
                        },
                        "cloneElement": [Function],
                        "createContext": [Function],
                        "createElement": [Function],
                        "createFactory": [Function],
                        "createRef": [Function],
                        "forwardRef": [Function],
                        "isValidElement": [Function],
                        "lazy": [Function],
                        "memo": [Function],
                        "useCallback": [Function],
                        "useContext": [Function],
                        "useDebugValue": [Function],
                        "useEffect": [Function],
                        "useImperativeHandle": [Function],
                        "useLayoutEffect": [Function],
                        "useMemo": [Function],
                        "useReducer": [Function],
                        "useRef": [Function],
                        "useState": [Function],
                        "version": "16.12.0",
                      }
                    }
                    auth={[Function]}
                    authActions={
                      Object {
                        "authorize": [Function],
                        "authorizeAccessCodeWithBasicAuthentication": [Function],
                        "authorizeAccessCodeWithFormParams": [Function],
                        "authorizeApplication": [Function],
                        "authorizeOauth2": [Function],
                        "authorizePassword": [Function],
                        "authorizeRequest": [Function],
                        "configureAuth": [Function],
                        "logout": [Function],
                        "preAuthorizeImplicit": [Function],
                        "showDefinitions": [Function],
                      }
                    }
                    authSelectors={
                      Object {
                        "authorized": [Function],
                        "definitionsForRequirements": [Function],
                        "definitionsToAuthorize": [Function],
                        "getConfigs": [Function],
                        "getDefinitionsByNames": [Function],
                        "isAuthorized": [Function],
                        "shownDefinitions": [Function],
                      }
                    }
                    configs={Object {}}
                    configsActions={
                      Object {
                        "loaded": [Function],
                        "toggle": [Function],
                        "update": [Function],
                      }
                    }
                    configsSelectors={
                      Object {
                        "get": [Function],
                      }
                    }
                    dispatch={[Function]}
                    err={[Function]}
                    errActions={
                      Object {
                        "clear": [Function],
                        "clearBy": [Function],
                        "newAuthErr": [Function],
                        "newSpecErr": [Function],
                        "newSpecErrBatch": [Function],
                        "newThrownErr": [Function],
                        "newThrownErrBatch": [Function],
                      }
                    }
                    errSelectors={
                      Object {
                        "allErrors": [Function],
                        "lastError": [Function],
                      }
                    }
                    fn={
                      Object {
                        "buildRequest": [Function],
                        "createXMLExample": [Function],
                        "execute": [Function],
                        "fetch": [Function],
                        "getScrollParent": [Function],
                        "inferSchema": [Function],
                        "memoizedCreateXMLExample": [Function],
                        "memoizedSampleFromSchema": [Function],
                        "opId": [Function],
                        "opsFilter": [Function],
                        "resolve": [Function],
                        "resolveSubtree": [Function],
                        "sampleFromSchema": [Function],
                        "sampleXmlFromSchema": [Function],
                        "serializeRes": [Function],
                        "shallowEqualKeys": [Function],
                      }
                    }
                    getComponent={[Function]}
                    getComponents={[Function]}
                    getConfigs={[Function]}
                    getState={[Function]}
                    getStore={[Function]}
                    getSystem={[Function]}
                    initOAuth={[Function]}
                    layout={[Function]}
                    layoutActions={
                      Object {
                        "changeMode": [Function],
                        "clearScrollTo": [Function],
                        "parseDeepLinkHash": [Function],
                        "readyToScroll": [Function],
                        "scrollTo": [Function],
                        "scrollToElement": [Function],
                        "show": [Function],
                        "updateFilter": [Function],
                        "updateLayout": [Function],
                      }
                    }
                    layoutSelectors={
                      Object {
                        "current": [Function],
                        "currentFilter": [Function],
                        "getScrollToKey": [Function],
                        "isShown": [Function],
                        "isShownKeyFromUrlHashArray": [Function],
                        "showSummary": [Function],
                        "urlHashArrayFromIsShownKey": [Function],
                        "whatMode": [Function],
                      }
                    }
                    log={[Function]}
                    makeMappedContainer={[Function]}
                    oas3={[Function]}
                    oas3Actions={
                      Object {
                        "setActiveExamplesMember": [Function],
                        "setRequestBodyValue": [Function],
                        "setRequestContentType": [Function],
                        "setResponseContentType": [Function],
                        "setSelectedServer": [Function],
                        "setServerVariableValue": [Function],
                      }
                    }
                    oas3Selectors={
                      Object {
                        "activeExamplesMember": [Function],
                        "requestBodyValue": [Function],
                        "requestContentType": [Function],
                        "responseContentType": [Function],
                        "selectedServer": [Function],
                        "serverEffectiveValue": [Function],
                        "serverVariableValue": [Function],
                        "serverVariables": [Function],
                      }
                    }
                    preauthorizeApiKey={[Function]}
                    preauthorizeBasic={[Function]}
                    render={[Function]}
                    spec={[Function]}
                    specActions={
                      Object {
                        "changeConsumesValue": [Function],
                        "changeParam": [Function],
                        "changeParamByIdentity": [Function],
                        "changeProducesValue": [Function],
                        "clearRequest": [Function],
                        "clearResponse": [Function],
                        "clearValidateParams": [Function],
                        "download": [Function],
                        "downloadConfig": [Function],
                        "execute": [Function],
                        "executeRequest": [Function],
                        "getConfigByUrl": [Function],
                        "invalidateResolvedSubtreeCache": [Function],
                        "logRequest": [Function],
                        "parseToJson": [Function],
                        "requestResolvedSubtree": [Function],
                        "resolveSpec": [Function],
                        "setMutatedRequest": [Function],
                        "setRequest": [Function],
                        "setResponse": [Function],
                        "setScheme": [Function],
                        "updateEmptyParamInclusion": [Function],
                        "updateJsonSpec": [Function],
                        "updateLoadingStatus": [Function],
                        "updateResolved": [Function],
                        "updateResolvedSubtree": [Function],
                        "updateSpec": [Function],
                        "updateUrl": [Function],
                        "validateParams": [Function],
                      }
                    }
                    specSelectors={
                      Object {
                        "allowTryItOutFor": [Function],
                        "basePath": [Function],
                        "canExecuteScheme": [Function],
                        "consumes": [Function],
                        "consumesOptionsFor": [Function],
                        "contentTypeValues": [Function],
                        "currentProducesFor": [Function],
                        "definitions": [Function],
                        "externalDocs": [Function],
                        "findDefinition": [Function],
                        "getLocalConfig": [Function],
                        "getParameter": [Function],
                        "hasHost": [Function],
                        "host": [Function],
                        "info": [Function],
                        "isOAS3": [Function],
                        "isSwagger2": [Function],
                        "lastError": [Function],
                        "loadingStatus": [Function],
                        "mutatedRequestFor": [Function],
                        "mutatedRequests": [Function],
                        "operationScheme": [Function],
                        "operationWithMeta": [Function],
                        "operations": [Function],
                        "operationsWithRootInherited": [Function],
                        "operationsWithTags": [Function],
                        "parameterInclusionSettingFor": [Function],
                        "parameterValues": [Function],
                        "parameterWithMeta": [Function],
                        "parameterWithMetaByIdentity": [Function],
                        "parametersIncludeIn": [Function],
                        "parametersIncludeType": [Function],
                        "paths": [Function],
                        "produces": [Function],
                        "producesOptionsFor": [Function],
                        "requestFor": [Function],
                        "requests": [Function],
                        "responseFor": [Function],
                        "responses": [Function],
                        "schemes": [Function],
                        "security": [Function],
                        "securityDefinitions": [Function],
                        "semver": [Function],
                        "servers": [Function],
                        "spec": [Function],
                        "specJson": [Function],
                        "specJsonWithResolvedSubtrees": [Function],
                        "specResolved": [Function],
                        "specResolvedSubtree": [Function],
                        "specSource": [Function],
                        "specStr": [Function],
                        "tagDetails": [Function],
                        "taggedOperations": [Function],
                        "tags": [Function],
                        "url": [Function],
                        "validateBeforeExecute": [Function],
                        "version": [Function],
                      }
                    }
                    state={
                      Immutable.Map {
                        "layout": Immutable.Map {
                          "layout": "BaseLayout",
                          "filter": null,
                        },
                        "spec": Immutable.Map {
                          "spec": "",
                          "url": "foo.example.com",
                          "loadingStatus": "loading",
                        },
                        "configs": Immutable.Map {},
                        "err": Immutable.Map {},
                        "auth": Immutable.Map {},
                        "oas3": Immutable.Map {},
                      }
                    }
                  >
                    <div
                      className="swagger-ui"
                    >
                      <div
                        className="loading-container"
                      >
                        <div
                          className="info"
                        >
                          <div
                            className="loading-container"
                          >
                            <div
                              className="loading"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </t>
                </r>
              </Connect(r)>
            </t>
          </r>
        </Connect(r)>
      </Provider>
    </r>
  </SwaggerUI>
</ActiveDocsSpec>
`;
